pilot/pkg/config/coredatamodel/syntheticserviceentrycontroller.go

```diff
@@ -0,0 +1,383 @@
+	if len(change.Objects) == 0 {
+		return nil
+	}
+
+	defer atomic.AddUint32(&c.synced, 1)
+
+	if change.Incremental {
+		// removed first
+		c.removeConfig(change.Removed)
+		c.incrementalUpdate(change.Objects)
+	} else {
+		c.configStoreUpdate(change.Objects)
+	}
+
+	return nil
+}
+
+// HasSynced returns true if the first batch of items has been popped
+func (c *SyntheticServiceEntryController) HasSynced() bool {
+	return atomic.LoadUint32(&c.synced) != 0
+}
+
+// RegisterEventHandler registers a handler using the type as a key
+func (c *SyntheticServiceEntryController) RegisterEventHandler(typ string, handler func(model.Config, model.Event)) {
+	log.Warnf("registerEventHandler: %s", errUnsupported)
+}
+
+// Run is not implemented
+func (c *SyntheticServiceEntryController) Run(stop <-chan struct{}) {
+	log.Warnf("run: %s", errUnsupported)
+}
+
+// Get is not implemented
+func (c *SyntheticServiceEntryController) Get(typ, name, namespace string) *model.Config {
+	log.Warnf("get %s", errUnsupported)
+	return nil
+}
+
+// Update is not implemented
+func (c *SyntheticServiceEntryController) Update(config model.Config) (newRevision string, err error) {
+	log.Warnf("update %s", errUnsupported)
+	return "", errUnsupported
+}
+
+// Create is not implemented
+func (c *SyntheticServiceEntryController) Create(config model.Config) (revision string, err error) {
+	log.Warnf("create %s", errUnsupported)
+	return "", errUnsupported
+}
+
+// Delete is not implemented
+func (c *SyntheticServiceEntryController) Delete(typ, name, namespace string) error {
+	log.Warnf("delete %s", errUnsupported)
+	return errUnsupported
+}
+
+func (c *SyntheticServiceEntryController) removeConfig(configName []string) {
+	if len(configName) == 0 {
+		return
+	}
+	c.configStoreMu.Lock()
+	defer c.configStoreMu.Unlock()
+
+	for _, fullName := range configName {
+		namespace, name := extractNameNamespace(fullName)
+		if byNamespace, ok := c.configStore[namespace]; ok {
+			if conf, ok := byNamespace[name]; ok {
+				delete(byNamespace, conf.Name)
+			}
+			// clear parent map also
+			if len(byNamespace) == 0 {
+				delete(byNamespace, namespace)
+			}
+		}
+	}
+}
+
+func (c *SyntheticServiceEntryController) convertToConfig(obj *sink.Object) (conf *model.Config, err error) {
+	namespace, name := extractNameNamespace(obj.Metadata.Name)
+
+	createTime := time.Now()
+	if obj.Metadata.CreateTime != nil {
+		if createTime, err = types.TimestampFromProto(obj.Metadata.CreateTime); err != nil {
+			log.Warnf("Discarding incoming MCP resource: invalid resource timestamp (%s/%s): %v", namespace, name, err)
+			return nil, err
+		}
+	}
+
+	conf = &model.Config{
+		ConfigMeta: model.ConfigMeta{
+			Type:              schemas.SyntheticServiceEntry.Type,
+			Group:             schemas.SyntheticServiceEntry.Group,
+			Version:           schemas.SyntheticServiceEntry.Version,
+			Name:              name,
+			Namespace:         namespace,
+			ResourceVersion:   obj.Metadata.Version,
```

Comment:
What's diff between the version here and `networking.alpha.istio.io/serviceVersion`/`"networking.alpha.istio.io/endpointsVersion"` ?

Reply:
The resource version is just the API version of the config, however service version and endpiont versions are bumped independently by the source server as the coressponding changes detected in each resource.